patches/components-image_fetcher-core-image_data_fetcher.cc.patch

```diff
@@ -0,0 +1,12 @@
+diff --git a/components/image_fetcher/core/image_data_fetcher.cc b/components/image_fetcher/core/image_data_fetcher.cc
+index 217f196fe7d1187060d78a1b8dc55352954f068e..e482b9da16d03a5cf98b0abf3a6f40ba6721f902 100644
+--- a/components/image_fetcher/core/image_data_fetcher.cc
++++ b/components/image_fetcher/core/image_data_fetcher.cc
+@@ -128,6 +128,7 @@ void ImageDataFetcher::FetchImageData(const GURL& image_url,
+   } else {
+     request->credentials_mode = network::mojom::CredentialsMode::kOmit;
+   }
++  BRAVE_ADJUST_HEADERS(request);
```

Comment:
I believe this should be called BRAVE_IMAGE_DATA_FETCHER_FETCH_IMAGE_DATA

Reply:
this is not the right place for this. As I said before we should add something similar to what we have for the profile network context to intercept this and add the headers based on an exact match.

Reply:
sorry, that's backwards, we need this in the profile network context similar to what we currently have for system

Reply:
this also needs tests

Reply:
https://github.com/brave/brave-core/blob/8d8f7e9ded2283015a31ac48ac651ead68303b56/browser/net/brave_service_key_network_delegate_helper.cc#L21

Reply:
existing test https://github.com/brave/brave-core/commit/c495ec97a2113fac758b6683c321819758b162f7#diff-069d6d661f7b05681e5af4572534f647977d85925e8f7503778d135356e78009